app/services/setting_service.rb

```diff
@@ -0,0 +1,70 @@
+require_relative "data_sources/json_setting_data_source"
+require_relative "../shared/logging_module"
+require_relative "../shared/available_settings"
+require_relative "./services"
+
+module FastlaneCI
+  class SettingServiceKeyNotFoundError < FastlaneCIError
+  end
+
+  # Provides access to system settings
+  class SettingService
+    include FastlaneCI::Logging
+    attr_accessor :setting_data_source
+
+    def initialize(setting_data_source: nil)
+      unless setting_data_source.nil?
+        unless setting_data_source.class <= SettingDataSource
+          raise "setting_data_source must be descendant of #{SettingDataSource.name}"
+        end
+      end
+
+      if setting_data_source.nil?
+        # Default to JSONSettingDataSource
+        # TODO: do we need `sample_data` here?
```

Comment:
No, I think we don't need a `sample_data` here. If there is a missing file, we should create it with default settings.

Reply:
Cool, we actually have the same code in all the other `_service.rb` files, I'll submit a separate PR that removes it from them also 👍

Reply:
Created https://github.com/fastlane/ci/issues/982